ludwig/data/preprocessing.py

```diff
@@ -1613,6 +1614,11 @@ def preprocess_for_training(
     if dataset is None and training_set is None:
         raise ValueError("No training data is provided!")
 
+    # preload ludwig datasets
+    dataset, training_set, validation_set, test_set = load_dataset_uris(
+        dataset, training_set, validation_set, test_set, backend
+    )
```

Comment:
Should we moe this one level above, i.e., at the `LudwigModel` level?

Reply:
Interesting thought. Looking at the code, `preprocess_for_training` is called from a few different places, so seems more general to keep it here to avoid being redundant.

Reply:
Sure, that seems fair